kernel/chown.c

```diff
@@ -229,14 +229,18 @@ long myst_syscall_fchownat(
     if ((flags & ~(AT_EMPTY_PATH | AT_SYMLINK_NOFOLLOW)) != 0)
         ERAISE(-EINVAL);
 
-    ECHECK(myst_get_absolute_path_from_dirfd(dirfd, pathname, flags, &abspath));
```

Comment:
While your changes to `myst_get_absolute_path_from_dirfd` easier to follow, but:

1. it forces the caller to allocate memory first, which resulted in a lot of changes to the many callers below. When a function is called many times, we tend to add complex logics to it rather than the call sites.
2. Some memory allocations could be otherwise saved now become mandatory.